src/osd/scheduler/mClockScheduler.cc

```diff
@@ -454,17 +468,61 @@ void mClockScheduler::enqueue(OpSchedulerItem&& item)
 
 void mClockScheduler::enqueue_front(OpSchedulerItem&& item)
 {
-  immediate.push_back(std::move(item));
+  unsigned priority = item.get_priority();
+  unsigned cutoff = get_io_prio_cut(cct);
+
+  if (priority >= cutoff) {
```

Comment:
In order for an item to be added to the immediate queue, the `priority >= cutoff` OR if the item is classified as op_scheduler_class::immediate. See `enqueue()` method on how to determine the classification.